File: tests/fixtures/extract/sample.py
class Server:
    def __init__(self, host: str, port: int):
        self.host = host
        self.port = port

    def start(self):
        self.listen()

    def listen(self):
        print(f"Listening on {self.host}:{self.port}")


def create_server(host: str, port: int) -> Server:
    return Server(host, port)

File: tests/fixtures/extract/sample.ts
interface Logger {
  log(message: string): void;
}

class Server {
  private host: string;
  private port: number;

  constructor(host: string, port: number) {
    this.host = host;
    this.port = port;
  }

  start(): void {
    this.listen();
  }

  private listen(): void {
    console.log(`Listening on ${this.host}:${this.port}`);
  }
}

function createServer(host: string, port: number): Server {
  return new Server(host, port);
}

const handleRequest = (req: string): void => {
  console.log(req);
};
